### PR TITLE
docs: Add 1.2.0 upgrade guide for Product Module

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: docs/content/upgrade-guides/modules/product/1-2-0.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# v1.2.0
+
+Version 1.2.0 of the Product Module introduces minor breaking changes to the `ProductService`.
+
+## Overview
+
+This release contains breaking change to the methods `list` and `listAndCount` in the `ProductService` of the Product Module.
+
+Specifically, the following category filter has changed:
```
Comment (shahednasser) [inline docs/content/upgrade-guides/modules/product/1-2-0.md]:
```suggestion
Specifically, the following category filter has changed:

```

File: docs/content/upgrade-guides/modules/product/1-2-0.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+```ts
```
Comment (shahednasser) [inline docs/content/upgrade-guides/modules/product/1-2-0.md]: I think it's fine to put the before and after in the same code block

Comment (shahednasser) [reply docs/content/upgrade-guides/modules/product/1-2-0.md]: Also it's kind of repetitive with the actions required right? I would just include the changes in the Actions Required section

File: docs/content/upgrade-guides/modules/product/1-2-0.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+npm install @medusajs/product@1.2.0
+```
+
+---
+
+## Actions Required
+
+### Change usage of Product Module
+
+If you've applied the category filter to the `list` or `listAndCount` methods of the Product Module, you will need to change it to be as follows:
```
Comment (shahednasser) [inline docs/content/upgrade-guides/modules/product/1-2-0.md]:
```suggestion
If you've applied the category filter to the `list` or `listAndCount` methods of the Product Module, you will need to change it to be as follows:

```

File: docs/content/upgrade-guides/modules/product/1-2-0.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+```ts
```
Comment (shahednasser) [inline docs/content/upgrade-guides/modules/product/1-2-0.md]: Would replace this with the before and after code block since it's clearer

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: docs/content/upgrade-guides/modules/product/1-2-0.md (hunk 2 above)
Comment (olivermrbl) [reply docs/content/upgrade-guides/modules/product/1-2-0.md]: Good point, will do!

**Comment by shahednasser** [comment]
Regarding the sidebar: You need to remove the `reverse` attribute and have a directory structure similar to the [plugins](https://github.com/medusajs/medusa/tree/develop/www/docs/content/upgrade-guides/plugins) directory (sorry for not mentioning this before 😅). You need to add the `_category_.json` file both in the `modules` directory and the `product` directory similar to plugins

**Review by shahednasser — APPROVED** [review]
LGTM

**Comment by shahednasser** [comment]
Oh can you run `yarn lint:content --fix` in the `www/docs` directory and commit the changes 😄

**Pull request merged**